butane_core/src/migrations/fsmigrations.rs

```diff
@@ -271,6 +271,17 @@ impl FsMigrations {
         f.write_all(serde_json::to_string(state)?.as_bytes())
             .map_err(|e| e.into())
     }
+    pub fn detach_latest_migration(&mut self) -> Result<()> {
+        let latest = self.latest().expect("There are no migrations");
```

Comment:
This line should return an error result if there are no migrations, not panic.

Reply:
Same for line 279

Reply:
This was hastily put together to help think through how detach/re-attach might work.  The PR will be abandoned if you disagree with the suggestion at https://github.com/Electron100/butane/issues/91#issuecomment-1567598015 of allowing people to edit the `.sql` files.  Without out that, the `unmake` approach is the better approach.

Reply:
Sorry, missed this reply. I agree with allowing people to edit the .sql files -- that's always been supported. The idea is that the migration system is as good as possible, but manual editing of the migration sql should be an escape hatch.